docs/user_guide/flyte_fundamentals/optimizing_tasks.md

```diff
@@ -52,13 +52,23 @@ represents the cache key. Learn more in the {ref}`User Guide <cache-offloaded-ob
 
 ## Retries
 
-Flyte also allows you to automatically retry failing tasks in the case of
-system-level or catastrophic errors that may arise from issues that don't have
-anything to do with user-defined code, like network issues and data center
-outages.
+Flyte allows you to automatically retry failing tasks in the case of system-level or catastrophic errors that may arise from issues unrelated to user-defined code, such as network issues and data center outages.
 
-The following version of the `compute_mean` task simulates these kinds of
-errors by randomly throwing a `RuntimeError` 5% of the time:
+### Understanding Error Types
+
+Flyte differentiates between two types of errors:
```

Comment:
this section feels disconnected from the retries section. instead of simply copying and pasting, could you please try focusing on making it flow naturally as a single, cohesive section?